util/http.py

```diff
@@ -236,6 +236,12 @@ def _request_with_timeout(cls, url, make_request_with, *args, **kwargs):
             if verbose:
                 logging.info("Sending %s request to %s: kwargs %r",
                              http_method, url, kwargs)
+            if len(args) == 1:
+                # requests.request takes two positional arguments,
+                # an HTTP method and a URL. In most cases, the URL
+                # gets added on here. But if you do pass in both
+                # arguments, it will still work.
+                args = args + (url,)
```

Comment:
This is the fix to the problem hidden by missing test coverage. `url` was being passed into this method and never actually being used. We were relying on `url` showing up _again_ in `*args`, which was confusing.